src/main/java/StressMain.java

```diff
@@ -649,6 +655,54 @@ private static Callable taskCallable(Workflow workflow, SolrCloud cloud, Map<Str
 				} catch (Exception ex) {
 					ex.printStackTrace();
 				}
+			} else if (type.moveReplica != null) {
+				long taskStart = System.currentTimeMillis();
+
+				String collectionName = params.get("COLLECTION");
+				String replicaName = "", fromNodeName = "";
+				try (CloudSolrClient client = new CloudSolrClient.Builder(List.of(cloud.getZookeeperUrl()), Optional.ofNullable(cloud.getZookeeperChroot())).build();) {
+					ClusterState state = client.getClusterStateProvider().getClusterState();
+					DocCollection collection = state.getCollection(collectionName);
+					List<Replica> replicas = collection.getReplicas();
+					// choose first replica since it's easy and should exist
+					Replica r = replicas.get(0);
+					replicaName = r.getName();
+					fromNodeName = r.getNodeName();
+				} catch (Exception e) {
+					log.warn("Exception occurred getting replicaName and fromNodeName: " + e.getMessage());
+				}
+
+				String targetNodeName = "";
+				// pick random node to move to
+				List<SolrNode> copyOfSolrNodes = new ArrayList<>(cloud.nodes);
+				Collections.shuffle(copyOfSolrNodes);
+				for (SolrNode sn : copyOfSolrNodes) {
+					String toNodeName = sn.getNodeName() + "_solr";
+					if (toNodeName != fromNodeName) {
+						targetNodeName = toNodeName;
+						break;
+					}
+				}
+				log.info(String.format("moving replica %s from %s to %s", replicaName, fromNodeName, targetNodeName));
+
+				try (HttpSolrClient client = new HttpSolrClient.Builder(cloud.nodes.get(0).getBaseUrl()).build();) {
+					SolrRequest request = new CollectionAdminRequest.MoveReplica(collectionName, replicaName, targetNodeName);
+					NamedList<Object> response = client.request(request);
+					log.info("MOVEREPLICA response: "+response.toString());
+				} catch (Exception e) {
+					log.warn("Exception occurred getting moving shard: " + e.getMessage());
+				}
+
+				long taskEnd = System.currentTimeMillis();
+				try {
+					String totalTime = String.valueOf((taskEnd-taskStart)/1000.0);
+
+					finalResults.get(taskName).add(Map.of("total-time", totalTime, "start-time", (taskStart- executionStart)/1000.0,
+							"end-time", (taskEnd- executionStart)/1000.0,
+							"init-timestamp", executionStart, "start-timestamp", taskStart, "end-timestamp", taskEnd));
+				} catch (Exception ex) {
```

Comment:
Let's not catch the exception here and let it bubble up. I know some of the older parts of the code logic still catch the exception but newer ones should just throw exception as is.

This allow the executor to properly detect the exception and skip all the follow tasks that depending on a failing tasks :)

Reply:
Oops, yep, I agree. I think I was doing some debugging here and just left this in.